### ISSUE TITLE
Compilation issues with nginx 1.24

### ISSUE THREAD

**Issue by sergey-dryabzhinsky** [wandenberg/nginx-push-stream-module#305]
OS: Ubuntu 16.04 amd64
nginx surce: 1.24
push-stream: 0.5.5

```
                 from /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src/ngx_http_push_stream_module.c:26:
src/core/ngx_array.h:32:1: note: expected ‘ngx_array_t * {aka struct <anonymous> *}’ but argument is of type ‘ngx_table_elt_t ** {aka struct ngx_table_elt_s **}’
 ngx_array_init(ngx_array_t *array, ngx_pool_t *pool, ngx_uint_t n, size_t size)
 ^
In file included from /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src/ngx_http_push_stream_module.c:29:0:
/home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src/ngx_http_push_stream_module_utils.c:2181:30: warning: passing argument 1 of ‘ngx_array_push’ from incompatible pointer type [-Wincompatible-pointer-types]
         ccp = ngx_array_push(&r->headers_out.cache_control);
                              ^
In file included from src/core/ngx_core.h:65:0,
                 from /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/include/ngx_http_push_stream_module.h:30,
                 from /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src/ngx_http_push_stream_module.c:26:
src/core/ngx_array.h:27:7: note: expected ‘ngx_array_t * {aka struct <anonymous> *}’ but argument is of type ‘ngx_table_elt_t ** {aka struct ngx_table_elt_s **}’
 void *ngx_array_push(ngx_array_t *a);
       ^
In file included from /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src/ngx_http_push_stream_module.c:29:0:
/home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src/ngx_http_push_stream_module_utils.c:2196:53: error: request for member ‘nelts’ in something not a structure or union
         for (i = 1; i < r->headers_out.cache_control.nelts; i++) {
                                                     ^
cc -c -fPIC -I/usr/include/luajit-2.0   -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fPIC -Wdate-time -D_FORTIFY_SOURCE=2 -I/opt/openssl111r/usr/include -L/opt/openssl111r/usr/lib -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fPIC -Wdate-time -D_FORTIFY_SOURCE=2 -I/opt/openssl111r/usr/include -L/opt/openssl111r/usr/lib -Wno-deprecated-declarations -DNDK_SET_VAR -std=gnu99 -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/http/modules/perl -I /usr/include -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-development-kit/objs -I objs/addon/ndk -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-development-kit/src -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-development-kit/objs -I objs/addon/ndk -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/src -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-push-stream-module/include -I /usr/include/luajit-2.0 -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/http-lua/src/api -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/http-nchan/src -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 -I /home/sergey/debuild/nginx/nginx-1.24/nginx/nginx-1.24.0/debian/modules/nginx-development-kit/src -I src/mail \
        -o objs/ngx_http_push_stream_module_modules.o \
        objs/ngx_http_push_stream_module_modules.c
objs/Makefile:1950: recipe for target 'objs/addon/src/ngx_http_push_stream_module.o' failed
make[2]: *** [objs/addon/src/ngx_http_push_stream_module.o] Error 1

```

**Comment by sergey-dryabzhinsky** [comment]
I'll check fix from #302

**Comment by sergey-dryabzhinsky** [comment]
Fix forking.
